httputil/request.go

```diff
@@ -172,14 +180,19 @@ func SetAuthType(req *http.Request, res *http.Response) {
 }
 
 func GetAuthType(res *http.Response) string {
-	auth := res.Header.Get("Www-Authenticate")
-	if len(auth) < 1 {
-		auth = res.Header.Get("Lfs-Authenticate")
-	}
 
-	if strings.HasPrefix(strings.ToLower(auth), "ntlm") {
-		return "ntlm"
+	for _, headerName := range []string{WwwAuthenticateHeader, LfsAuthenticateHeader} {
+		authHeaders := res.Header[headerName]
```

Comment:
I C# code, I like to create temp variables in cases like this - it makes debugging easier in some cases. For example, I can set breakpoint when variable is set to specific value. I remove temp variable.